lib/govuk_schemas/random_item_generator.rb

```diff
@@ -10,8 +10,9 @@ module GovukSchemas
   #
   # @private
   class RandomItemGenerator
-    def initialize(schema:)
+    def initialize(schema:, seed: nil)
       @schema = schema
+      srand(seed) unless seed.nil?
```

Comment:
If I understand correct this covers the whole application this is used in and is not isolated to this class. 

It'd probably be less surprising if this was done in govuk_publishing_components itself as that is the place that we wish to make random number generation deterministic.

Reply:
Thanks for the review, Kevin. Agreed - I wasn't sure if Ruby does anything magic in terms of scoping the `srand` to the module wherein it is called. But if it's a global thing, it's better that that is defined further upstream.

You rightly point out that the removal of the SecureRandom code is still required in govuk_schemas, and can be kept separate from any seed logic. I'll work to refactor that out 👍

Reply:
It might be nice if we could refactor this code to use [the `Random` class](https://ruby-doc.org/core-2.6/Random.html), which takes a `seed` as an argument instead rather than using the global random number generator.

Reply:
I was looking at the Random class prior to this PR, but it looked like it would be quite a rewrite, especially if we wanted to litter the seed value throughout. The current iteration makes no assumptions about its environment, it just no longer _prevents_ you from setting a global seed via `srand`, which seems a good step for now.